src/librbd/AioObjectRequest.cc

```diff
@@ -410,27 +451,10 @@ namespace librbd {
           		       << m_object_off << "~" << m_object_len << dendl;
         m_state = LIBRBD_AIO_WRITE_PRE;
 
-        uint8_t new_state;
```

Comment:
I guess now WRITE_OBJECT_MAP is analogous to WRITE_PRE, though the state diagram isn't exactly the same. We now skip the object map update in the copyup case (because we don't know yet if it's needed) and handle it as part of the copyup operation, if necessary.